framework/src/org/checkerframework/framework/flow/CFCFGBuilder.java

```diff
@@ -144,6 +148,16 @@ protected VariableTree createEnhancedForLoopArrayVariable(
             factory.shouldCache = false;
             AnnotatedTypeMirror annotatedArrayType = factory.getAnnotatedType(expression);
             factory.shouldCache = oldShouldCache;
+            if (annotatedArrayType.getKind() == TypeKind.WILDCARD) {
+                if (((AnnotatedWildcardType) annotatedArrayType).isUninferredTypeArgument()) {
+                    TypeMirror type = TreeUtils.typeOf(expression);
+                    AnnotatedArrayType newArrayType =
+                            (AnnotatedArrayType)
+                                    AnnotatedTypeMirror.createType(type, factory, false);
+                    newArrayType.setComponentType(annotatedArrayType);
```

Comment:
Should the array type get any annotations?
The Nullness Checker e.g. ensures that we only do an enhanced-for over a non-null array. What defaults get applied to this type?